Clarify U-mode deprecation in open()

ncoghlan opened the pull request:
The previous wording could be read as saying that universal
newlines mode itself was deprecated, when it's only the 'U'
character in the mode field that should be avoided.

ncoghlan committed:
Clarify U-mode deprecation in open()

The previous wording could be read as saying that universal
newlines mode itself was deprecated, when it's only the 'U'
character in the mode field that should be avoided.

ncoghlan committed:
Too much Markdown lately :)

eamanu commented on Doc/library/functions.rst:
```diff
@@ -1004,7 +1004,7 @@ are always available.  They are listed here in alphabetical order.
    ``'t'``   text mode (default)
    ``'+'``   open a disk file for updating (reading and writing)
-   ``'U'``   :term:`universal newlines` mode (deprecated)
+   ``'U'``   equivalent to ``newline=None`` (deprecated, use *newline* option)
```
For me is a good improved. How say @csabella, it would be great have a link to *newline* definition

willingc replied:
Perhaps it would be clearer to put `:deprecated:` before the rest of the text
```
``'U'``   :deprecated: Use ``newline`` option instead. Equivalent to ``newline=None``
```

Having a link as suggested by @csabella would be great.

willingc reviewed (commented):
This would probably be a bit clearer by putting the deprecation at the start of the line. Thanks.

ncoghlan committed:
Further adjustments to deprecation notice

- add cross-reference to newline parameter description
- move deprecation warning to the start of the paragraph
- explain why the mode is deprecated (i.e. it's the default for
  test mode now,  so specifying it explicitly is redundant)

ncoghlan committed:
Fix table markup

ncoghlan committed:
Move 'U' entirely out of file mode table

I couldn't get the longer entry to look any good inside the table, so
I moved it out to a separate paragraph after the description of the
actually useful file modes.

willingc reviewed (approved):
Nice! Thanks @ncoghlan
